libcryptsetup-rs-sys/build.rs

```diff
@@ -24,7 +27,16 @@ fn build_safe_free() {
 }
 
 fn generate_bindings(safe_free_is_needed: bool) {
-    let builder = bindgen::Builder::default().header("header.h").size_t_is_usize(true);
+    match pkg_config::probe_library("libcryptsetup") {
```

Comment:
Could we unify the library detection? This would involve taking:
```
 Config::new()
        .atleast_version("2.2.0")
        .probe("libcryptsetup")
```
and putting it in `main()` and then modifying all methods that require the use of the `Library` data structure to take a parameter.

This would allow us to only probe for the library a single time when compiling which I think would be better.